src/bot.ts

```diff
@@ -1,17 +1,35 @@
 // deno-lint-ignore-file camelcase
-import { BotError, Composer, run } from "./composer.ts";
+import { BotError, Composer, type Middleware, run } from "./composer.ts";
 import { Context } from "./context.ts";
 import { Api } from "./core/api.ts";
 import {
     type ApiClientOptions,
     type WebhookReplyEnvelope,
 } from "./core/client.ts";
 import { GrammyError, HttpError } from "./core/error.ts";
+import { type Filter, type FilterQuery, parse, preprocess } from "./filter.ts";
 import { debug as d } from "./platform.deno.ts";
 import { type Update, type UserFromGetMe } from "./types.ts";
 const debug = d("grammy:bot");
+const debugWarn = d("grammy:warn");
 const debugErr = d("grammy:error");
 
+export const DEFAULT_UPDATE_TYPES = [
```

Comment:
All update types appear to be available in `filter.ts` as `UPDATE_KEYS` already. Should we prefer to do

```ts
export const ALL_UPDATE_TYPES = Object.keys(UPDATE_KEYS);
export const DEFAULT_UPDATE_TYPES = ALL_UPDATE_TYPES.filter(u => u !== 'chat_member');
```

To keep a single source of truth for update types in grammY's source?

Reply:
I considered this, but they're semantically unrelated because the supported filter queries need not be the same as the Bot API specs. That's why I didn't follow DRY.

Reply:
Is it completely semantically unrelated, or perhaps just that it is a superset of Bot API specs? I'd presume you always want to support update types from Bot API specs, but want room to also conjure L1s of your own.

If so, we can probably do TG_UPDATE_KEYS and CUSTOM_UPDATE_KEYS (if and when it is needed); where filter.ts exports the former, and filters on the set union of the above?

Reply:
> Is it completely semantically unrelated

Yes. I don't want to mix those two things. One is a mirror of the specs, and the other is the base for filter query validation. We don't know if we want to have filter queries for all fields in updates, or if we will have excessive queries that are not found in updates (shortcuts that aren't computed perhaps).